### PR TITLE
fix(DB/Karazhan): Skeletal Usher - Ice Tomb's target

### LINKED ISSUE #16031: Karazhan - Trash - Skeletal Usher
Original issue: https://github.com/chromiecraft/chromiecraft/issues/4866

### What client do you play on?

enUS

### Faction

Both

### Content Phase:

70

### Current Behaviour

They are not casting Ice Tomb to the target highest on threat, but randomly
https://youtu.be/px3WCMXzK_Q?t=3682

### Expected Blizzlike Behaviour

They should cast Ice Tomb on the target with highest threat, then moving to the 2nd on threat, usually the Off Tank.

### Source

https://wowpedia.fandom.com/wiki/Skeletal_Usher

**Wowhead:**
https://www.wowhead.com/wotlk/npc=16471/skeletal-usher#abilities;mode:n10

### Steps to reproduce the problem

1. Enter Karazhan
2. Head to Opera by going upstairs, then take a right and into the stairs, then straight.
3. Fight the Skeleton Usher

### Extra Note

usher

### AC rev. hash/commit

49820e475e63

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [mod-zone-difficulty](https://github.com/azerothcore/mod-zone-difficulty)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-event-scripts](https://github.com/55Honey/Acore_eventScripts)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by Si1ker**
## Issues Addressed:

-  Closes https://github.com/chromiecraft/chromiecraft/issues/4866
-  Closes #16031

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Nyeriah** [comment]
The issue I had with this and why I didn’t address it before is that the report says they should pick another target if the main target is frozen, I’m not sure if that’s achievable with SAI

**Comment by Si1ker** [comment]
> The issue I had with this and why I didn’t address it before is that the report says they should pick another target if the main target is frozen, I’m not sure if that’s achievable with SAI

What I've read from comments is how to "cheat" them by let the tanks go first, build aggro - get Ice Tomb'ed then go dps 'till death. Some comments also mention Tank and Off-Tank strat for the main to get ice tomb'ed and Off-Tank to retake it (without any DPS dealing damage).

TLDR: Probably link Ice Tomb SAI entry with another one with SMART_ACTION_THREAT_SINGLE_PCT to aggroReset ?

Thoughts?

source: https://www.wowhead.com/wotlk/npc=16471/skeletal-usher#comments:id=2828611
image

**Comment by Gultask** [comment]
There's a lot of creatures that changes targets based on the victims status. Conflagration and CONFUSE statuses, for example, always puts the affected player at the bottom of the threat list. I've seen it sometimes with STUN auras as well, which is the case here. I think it's more of an core AI issue than SAI one.

**Comment by Si1ker** [comment]
> 

Hmm, seems a reasonable point, closing this and you file the issue as Core Generic? or do we all agree this is a good "partial" fix to at least make 50% of the mechanic?

**Comment by Si1ker** [comment]
> There's a lot of creatures that changes targets based on the victims status. Conflagration and CONFUSE statuses, for example, always puts the affected player at the bottom of the threat list. I've seen it sometimes with STUN auras as well, which is the case here. I think it's more of an core AI issue than SAI one.

Hmm, that seems a reasonable point, closing this and you file the issue as Core Generic? or do we all agree this is a good "partial" fix to at least make 50% of the mechanic?
